src/bootstrap/src/core/build_steps/dist.rs

```diff
@@ -421,13 +421,13 @@ impl Step for Rustc {
                 builder.install(&rustdoc, &image.join("bin"), FileType::Executable);
             }
 
+            let ra_proc_macro_srv_compiler =
+                builder.compiler_for(compiler.stage, builder.config.build, compiler.host);
+            builder.ensure(compile::Rustc::new(ra_proc_macro_srv_compiler, compiler.host));
```

Comment:
Doesn't this now build the compiler for the proc macro server even if `tool::RustAnalyzerProcMacroSrv` should not actually be built? :thinking: It's not always enabled.

Reply:
Compiler is passed before the condition here so I'm not sure how to guard against that without applying ugly hacks. Also, in practice, I don't think we ever hit the case of "trying to build `RustAnalyzerProcMacroSrv` while the compiler isn't compiled/ready for use".

Reply:
Ok, fair enough.